### ISSUE TITLE
Replace legacy format with lineage

### ISSUE THREAD

**Issue by bentsherman** [nextflow-io/nf-prov#16]
The original output format created for this plugin could be improved in a number of ways, including:
- list pipeline params
- include task scripts

The format is currently called "legacy", but I would like to rename it to something like "prov" or "manifest" and refactor it. The legacy format can always be used with v1.0.

I was going to remove it, but now I think it would be useful to have a format that we can customize to our needs, since the BCO and RO crate standards might not include everything that people want.

**Comment by stevekm** [comment]
I was wondering if it might be possible to list some of the desired fields to include in this format's output. Similar to how the Nextflow Trace lets you configure the fields to include.

**Comment by bentsherman** [comment]
Yes I think that would be a good way to control the level of detail

**Comment by bentsherman** [comment]
The [data lineage](https://nextflow.io/docs/latest/tutorials/data-lineage.html) feature in Nextflow would be a good candidate to replace this legacy format. Basically, collect all lineage records for a workflow run into a single JSON report.

On the other hand, that will become very large as the lineage history grows. Honestly it seems like an anti-pattern. You can already explore the lineage data via `nextflow lineage` which is a much better experience. The only reason to generate a report is for standard formats like BCO / WRROC for submitting to regulatory bodies.
